v2/contribute/decisions/thirdparty/0008-default-tenant-creation-in-core.mdx

```diff
@@ -0,0 +1,42 @@
+---
+id: "0008"
+title: Creation of default tenant in core
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Creation of default tenant in core
+
+<DecisionHeader status="proposed" lastUpdate="2022-10-25" created="2022-10-25" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+Since the recipe functions/APIs query the core if the recipe is enabled in core, what is the desired behaviour for the default tenant
+
+## Considered Options
+
+* Do not create a default tenant in core, assume all recipes are enabled if config not found
+* Create a default tenant in core with all recipes (emailpassword, passwordless and thirdparty) enabled
+
+
+## Decision Outcome
+
+Chosen option: **Create a default tenant in core with all recipes enabled**, because
+
+* While adding thirdparty providers to the core, there is no ambiguity if the recipe should be enabled at that time
+* Recipe functions/APIs can check if the recipe is enabled without having to handle the corner cases
+
+## Pros and Cons of the Options
+
+### Do not create a default tenant in core, assume all recipes are enabled if config not found
+
+<ArgumentCon> Might unintentionally enable all recipes for an unknown tenant, for which config won't be found </ArgumentCon>
+
+### Create a default tenant in core with all recipes enabled
+
+<ArgumentNeut> Safe default for the default tenant </ArgumentNeut>
+<ArgumentPro> No ambiguity when adding thirdparty providers, as the enabled flag is already set to true </ArgumentPro>
```

Comment:
how is this pro point even related to this ADR? 

The reason for creating a default tenant in the core is so that by default, for single tenant all the recipes will work if they are initialised by the dev - which is the expected dev experience. If we don't make a default tenant, we will have to have edge case checks everywhere to enable recipes if no tenant exists and if the user is using single tenancy.